test/libsolidity/semanticTests/abiEncoderV1/abi_encode_empty_string.sol

```diff
@@ -0,0 +1,31 @@
+contract C {
+    function f1() public returns (bytes memory) {
+        return abi.encode("");
+    }
+    function f2(string calldata msg) public returns (bytes memory) {
+        return abi.encode(msg);
+    }
+    function g1() public returns (bytes memory) {
+        return abi.encodePacked("");
+    }
+    function g2(string calldata msg) public returns (bytes memory) {
+        return abi.encodePacked(msg);
+    }
+    function h1() public returns (bytes memory) {
+        return abi.encodeWithSignature("(string)", "");
+    }
+    function h2(string calldata msg) public returns (bytes memory) {
+        return abi.encodeWithSignature("(string)", msg);
+    }
+}
+
+// ====
+// ABIEncoderV1Only: true
+// compileViaYul: also
+// ----
+// f1() -> 32, 0x40, 0x20, 0
+// f2(string): "" -> 32, 0x40, 0x20, 0
```

Comment:
this is not the encoding of the empty string. you need `0x20, 0, ""`. Whether or not to put the `""` here is the big question.

Reply:
I know, but left this short input too, because V1 accepts it.